crates/uv-resolver/src/resolver/mod.rs

```diff
@@ -3541,6 +3599,35 @@ fn find_conflicting_extra(conflicting: &Conflicts, reqs: &[Requirement]) -> Opti
     None
 }
 
+/// Compute the set of markers for which a package is known to be relevant.
+fn find_environments(id: Id<PubGrubPackage>, state: &State<UvDependencyProvider>) -> MarkerTree {
+    let package = &state.package_store[id];
+    if package.is_root() {
+        return MarkerTree::TRUE;
+    }
+
+    // Retrieve the incompatibilities for the current package.
+    let Some(incompatibilities) = state.incompatibilities.get(&id) else {
+        return MarkerTree::FALSE;
+    };
+
+    // Find all dependencies on the current package.
+    let mut marker = MarkerTree::FALSE;
+    for index in incompatibilities {
```

Comment:
Don't we add additional incompatibilities that target `id` when we find a new package `foo` with `foo --[marker]--> id`? Could that lead to packages missing platform support because we could have a case like the following:

Required platform linux

Choose A 2, with `A -> B; windows`, `A -> C`
Choose B 2. B 2 does have only a mac wheel but that's fine because it's only required on windows, so our required linux is coverd.
Choose C 2, with `C -> B`

Try to install on linux. We try to install A 2, C 2 and B 2, but B 2 fails since it has no wheel.

Reply:
@konstin -- I think this would actually work as-is, though for "bad" reasons. Basically, when we visit `B`, we won't know that it's "only required for Windows", because we don't do "aggressive forking" today, so we'd still try to solve for Linux. It might get filtered out later? I can't quite remember.

Reply:
Could you add a test case covering this scenario? So it doesn't fail with future resolver changes.